conversion/conversion.go

```diff
@@ -165,12 +166,61 @@ func getDbNameFromSQLConnectionStr(driver, sqlConnectionStr string) string {
 	return ""
 }
 
+func getInfoSchemaForShard(shardConnInfo profiles.DirectConnectionConfig, driver string, targetProfile profiles.TargetProfile) (common.InfoSchema, error) {
+	params := make(map[string]string)
+	params["host"] = shardConnInfo.Host
+	params["user"] = shardConnInfo.User
+	params["dbName"] = shardConnInfo.DbName
+	params["port"] = shardConnInfo.Port
+	params["password"] = shardConnInfo.Password
+	sourceProfileConnectionMySQL, err := profiles.NewSourceProfileConnectionMySQL(params)
+	if err != nil {
+		return nil, fmt.Errorf("cannot parse connection configuration for the primary shard")
+	}
+	sourceProfileConnection := profiles.SourceProfileConnection{Mysql: sourceProfileConnectionMySQL, Ty: profiles.SourceProfileConnectionTypeMySQL}
```

Comment:
Should add a not on what would change while adding other source databases

Reply:
Added.